app/views/active_storage/blobs/_blob.html.erb

```diff
@@ -1,14 +1,35 @@
-<figure class="attachment attachment--<%= blob.representable? ? "preview" : "file" %> attachment--<%= blob.filename.extension %>">
-  <% if blob.representable? %>
-    <%= image_tag blob.representation(resize_to_limit: local_assigns[:in_gallery] ? [ 800, 600 ] : [ 1024, 768 ]) %>
+<% options ||= {} %>
+
+<% if blob.representable? %>
+  <%
+    image_source = nil
+    if blob.content_type == "application/pdf"
+      # If it's a PDF, increase the DPI to make the preview image readable
+      options[:width] ||= 400
+      options[:height] ||= 400
+      preview_options = { resize_to_limit: [options[:width], options[:height]] }
+      preview_options[:saver] = { dpi: 300 }
+      image_source = blob.representation(preview_options)
+    else
+      image_source = blob
+    end
+  %>
+
+  <%= link_to url_for(blob), target: "_blank", title: "View original: #{blob.filename}", class: "inline-block" do %>
+    <%= image_tag image_source, options.merge(class: "rounded-md max-w-full h-auto") %>
   <% end %>
```

Comment:
We're definitely going to need to do something different here. I think we'll still need to use `blob.representation` for _most_ media formats. Not calling `representation` means that we're linking directly to the uploaded assets instead of creating a thumbnail. For gifs, this is (probably) a good thing, but for huge jpegs it's definitely not good. We may want to give app developers some easy way to toggle whether or not we embed the full gif or a shot of the first frame. I think `representable?` might cover more media formats than _only_ images, so we'll definitely need `representation` for those.